lib/core/src/Cardano/Wallet/Api/Types.hs

```diff
@@ -969,8 +971,20 @@ data ByronWalletPutPassphraseData = ByronWalletPutPassphraseData
     , newPassphrase :: !(ApiT (Passphrase "user"))
     } deriving (Eq, Generic, Show)
 
+data ApiSealedTxEncoding = HexEncoded | Base64Encoded
+      deriving (Eq, Generic, Show)
+      deriving anyclass NFData
+
+instance ToJSON ApiSealedTxEncoding where
+    toJSON HexEncoded = Bool True
+    toJSON Base64Encoded = Bool False
+instance FromJSON ApiSealedTxEncoding where
+    parseJSON = withBool "ApiSealedTxEncoding" $ \b -> case b of
+        True -> pure HexEncoded
+        False -> pure Base64Encoded
+
 data ApiConstructTransaction (n :: NetworkDiscriminant) = ApiConstructTransaction
-    { transaction :: !(ApiT SealedTx)
+    { transaction :: !(ApiT SealedTx, ApiSealedTxEncoding)
     , coinSelection :: !(ApiCoinSelection n)
     , fee :: !(Quantity "lovelace" Natural)
     } deriving (Eq, Generic, Show, Typeable)
```

Comment:
```
data ApiSerialisedTransaction = ApiSerialisedTransaction
    { serialisedTxSealed :: ApiT SealedTx
    , serialisedTxEncoding :: ApiSealedTxEncoding
    } deriving stock (Eq, Generic, Show)
      deriving anyclass (NFData)
```
I find this representation more readable at call sites because of the names of the accessor functions, as opposed to meaningless `fst` / `snd`, example:
this line is less readable:
```haskell
let bytes = serialisedTx $ getApiT $ fst $ (tx ^. #transaction)
```
than this one:
```haskell
let bytes = serialisedTx $ getApiT $ serialisedTxSealed $ tx ^. #transaction